frontend/src/components/pages/HomePage.tsx

```diff
@@ -1,31 +1,133 @@
+import React, { useContext, useState } from "react";
+import { DocumentNode, gql, useQuery } from "@apollo/client";
 import StoryList from "../story/StoryList";
 import { StoryCardProps } from "../story/StoryCard";
+import { Role } from "../../constants/Enums";
+import AuthContext from "../../contexts/AuthContext";
+import ToggleButton from "../navigation/ToggleButton";
+import Header from "../navigation/Header";
+import "./HomePage.css";
 
-const GET_STORIES = gql`
-  query GetStories {
-    stories {
-      id
-      title
-      description
-      youtubeLink
-      level
-    }
-  }
-`;
+type QueryInformation = {
+  fieldName: string;
+  string: DocumentNode;
+};
 
 const HomePage = () => {
+  // TODO: ow do private methods exist?
+  const buildHomePageStoriesQuery = (
+    displayMyStories: boolean,
+    translationLanguage: string,
+    role: Role,
+    level: number,
+    userId: number,
+  ): QueryInformation => {
+    if (role === Role.Translator && !displayMyStories) {
+      return {
+        fieldName: "storiesAvailableForTranslation",
+        string: gql`
+          query {
+            storiesAvailableForTranslation(
+              language: "${translationLanguage}",
+              level: ${level}
+            ) {
+              storyId: id
+              title
+              description
+              youtubeLink
+              level
+            }
+          }
+        `,
+      };
+    }
+
+    if (role === Role.Reviewer && !displayMyStories) {
+      return {
+        fieldName: "storyTranslationsAvailableForReview",
+        string: gql`
+          query {
+            storyTranslationsAvailableForReview(
+              language: "${translationLanguage}",
+              level: ${level}
+            ) {
+              storyId
+              storyTranslationId
+              title
+              description
+              youtubeLink
+              level
+            }
+          }
+        `,
+      };
+    }
+
+    return {
+      fieldName: "storyTranslationsByUser",
+      string: gql`
+        query {
+          storyTranslationsByUser(
+            userId: ${userId},
+            translator: ${role === Role.Translator}
+          ) {
+            storyId
+            storyTranslationId
+            title
+            description
+            youtubeLink
+            level
+          }
+        }
+      `,
+    };
+  };
+
+  const [displayMyStories, setDisplayMyStories] = useState<boolean>(true);
+  const [translationLanguage, setTranslationLanguage] = useState<string>(
+    "RUSSIAN",
+  );
+  const [role, setRole] = useState<Role>(Role.Translator);
+  const [maxLevel, setMaxLevel] = useState<number>(1);
   const [stories, setStories] = useState<StoryCardProps[] | null>(null);
 
-  useQuery(GET_STORIES, {
+  const { authenticatedUser } = useContext(AuthContext);
+
+  // TODO: dummy setting to avoid the unused vars error :(
+  if (authenticatedUser == null) {
+    setDisplayMyStories(true);
```

Comment:
to avoid linting issues for currently unused state setters, I believe you can skip declaring them for now + remove this if statement. ie:
`const [role] = useState<Role>(Role.Translator);`